emscripten.py

```diff
@@ -163,7 +163,7 @@ def update_settings_glue(wasm_file, metadata):
     # callMain depends on this library function
     settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$stringToUTF8OnStack']
 
-  if settings.STACK_OVERFLOW_CHECK and not settings.SIDE_MODULE:
+  if settings.STACK_OVERFLOW_CHECK == 1 and not settings.SIDE_MODULE:
```

Comment:
Why exclude mode 2 here? Mode 2 does all that 1 does + more, according to the `settings.js` docs, which I think is a behavior we should keep.

Reply:
I'm not sure its necessary to use the stack cookie check if we are using full stack checking (`STACK_OVERFLOW_CHECK == 2`).

Its useful here for to do it this way as it then allows for `-sSTACK_OVERFLOW_CHECK=2` to work with `-sSTACK_FIRST` (since we want to disable the cookie check in the later case.. so I don't want =2 to imply =1).  If you feel strongly I can work around this using a new internal setting.. something like `STACK_COOKIE_CHECKS`